src/NHibernate/Linq/Visitors/ResultOperatorProcessors/ProcessFetch.cs

```diff
@@ -11,12 +12,15 @@ public class ProcessFetch
 	{
 		public void Process(FetchRequestBase resultOperator, QueryModelVisitor queryModelVisitor, IntermediateHqlTree tree)
 		{
-			var querySource = QuerySourceLocator.FindQuerySource(
-				queryModelVisitor.Model,
-				resultOperator.RelationMember.DeclaringType);
-			var name = queryModelVisitor.VisitorParameters.QuerySourceNamer.GetName(querySource);
+			var lastFromNode = tree.GetLastFromChildNode();
```

Comment:
Why have you decided to find the from with HQL? `QuerySourceSelector` (which is currently an inner class of `Joiner`) can reliably find the correct query source.